.github/workflows/image-build.yaml

```diff
@@ -27,11 +30,83 @@ jobs:
       - name: Archive bpfman Binaries
         uses: actions/upload-artifact@v4
         with:
-          name: bpfman
+          name: bpfman-for-build-args
           path: |
             ./target/debug/bpfman.tar.gz
 
-  build-and-push-images:
+  # Build bpfman for each architecture and store the bpfman binaries in artifacts
+  # for the "build-and-push-userspace-images" step. The rust tool `cross` cannot be run in a
+  # container, so build the binaries locally, and the container build will just copy
+  # the files into the container.
+  build-bpfman-per-arch:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write # needed for signing the images with GitHub OIDC Token
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - arch: amd64
+            rust-target: x86_64-unknown-linux-gnu
+            filename: linux-x86_64
+            command: cargo
+          - arch: arm64
+            rust-target: aarch64-unknown-linux-gnu
+            filename: linux-arm64
+            command: cross
+          - arch: ppc64le
+            rust-target: powerpc64le-unknown-linux-gnu
+            filename: linux-ppc64le
+            command: cross
+          - arch: s390x
+            rust-target: s390x-unknown-linux-gnu
+            filename: linux-s390x
+            command: cross
+
+    name: Build bpfman (bpfman-${{ matrix.arch.arch }})
+    steps:
+      - name: Checkout bpfman
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain - stable
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          target: ${{ matrix.arch.rust-target }}
+
+      # Only install cross if we need it
+      - name: Install Cross
+        if: ${{ matrix.arch.command == 'cross' }}
+        shell: bash
+        run: |
+          cargo install cross --git https://github.com/cross-rs/cross
+
+      # BILLY: Is this needed?
```

Comment:
hopefully u know by now if this needed or not ?

Reply:
Not yet. Quick glance it supposed to speed up rust builds with some caching, but not sure of the details. Need to run with it enabled and disabled and see if it speeds anything up or not. Just haven't gotten to it yet.

Reply:
Done